solutions/cses-1091.mdx

```diff
@@ -51,3 +56,69 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+
+<JavaSection>
+<Warning>
+Due to the way CSES's grader works (allowing 1 second for all problems + being slow), test cases 5 and 8 TLE (time limit exceeded). USACO gives Java test cases 4 seconds, so this isn't representative of what competing with Java will be like. 
+	
+Also, Java doesn't have a defined multiset function, so we have to use an ordered map with some special modifications, using values as a counter of the number of identical sets. This allows us to distinguish between sets, and remove each set once.
+</Warning>
+```java
+import java.io.*;
+import java.util.*;
+
+public class ConcertTickets {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+
+		int ticketNum = io.nextInt();
+		int peopleNum = io.nextInt();
+		// No multiset in java so we'll have to use a TreeMap
+		NavigableMap<Integer, Integer> ticketMultiset = new TreeMap<>();
+		Map.Entry<Integer, Integer> val;
+
+		for (int i = 0; i < ticketNum; i++) {
+			int priceOfTicket = io.nextInt();
+			/*
+			 * If the set already contains the same key value, create another pair 
+			 * next to it
+			 */
+			if (ticketMultiset.containsKey(priceOfTicket)) {
+				ticketMultiset.put(priceOfTicket, ticketMultiset.get(priceOfTicket) + 1);
+			} else {
+				ticketMultiset.put(priceOfTicket, 1);
+			}
+		}
+
+		for (int i = 0; i < peopleNum; i++) {
+			int customerMaxPrice = io.nextInt();
+			/*
+			 * Get the optimal ticket for this customer 
+			 * We add 1 because lowerEntry returns the highest *strictly* lower entry
+			 */
+			val = ticketMultiset.lowerEntry(customerMaxPrice + 1); 
+			// If there are no possible prices, we can exit and return -1
+			if (val != null) {
+				io.println(val.getKey());
+				/*
+				 * If there's more than one set with the key value, then replace
+				 * the current set with the next-lowest set with the same key 
+				 */
+				if (val.getValue() == 1) {
+					ticketMultiset.remove(val.getKey());
+
+				} else {
+					ticketMultiset.put(val.getKey(), val.getValue() - 1);
+				}
+			} else
+				io.println(-1);
+		}
```

Comment:
either add brackets to this else statement or put the `io.println(-1)` on the same line as `else`